chef/cookbooks/network/recipes/default.rb

```diff
@@ -222,6 +228,13 @@ def get_datapath_id_for_ovsbridge(bridge)
       # interface is brought back up.
       unless bond.slaves.include? i
         ::Kernel.system("wicked ifdown #{i.name}")
+        # Delete the associated config file as well. Otherwise, in cases where
+        # this recipe fails (for whatever reason) before writing the new
+        # configuration files, a "network restart", as e.g. issued by crowbar_join
+        # will endup with an inconsistent configuration.
+        if ::File.exist?("/etc/sysconfig/network/ifcfg-#{i.name}")
+          ::File.delete("/etc/sysconfig/network/ifcfg-#{i.name}")
+        end
```

Comment:
Shouldn't we use some similar code as the one in `kill_nic`? This would make it work cross-distro, but also remove some additional files that may exist.

Reply:
I considered `kill_nic`, but AFAICS it does a few thing too much here. Regarding cross-distro, I must admit that I ignored that aspect as we already have that `wicked` call just a few lines above.

Reply:
But you're right I forgot about the other possible files ...

Reply:
I was thinking about splitting `kill_nic` and create something like `kill_nic_files`.

Reply:
@vuntz Ok. I've adjusted the code accordingly.